engines/director/score.cpp

```diff
@@ -462,15 +471,18 @@ void Score::update() {
 		}
 	}
 
-	byte tempo = _frames[_currentFrame]->_tempo;
-	if (tempo) {
+	byte tempo = _frames[_currentFrame]->_scoreCachedTempo;
+	// puppetTempo is overridden by changes in score tempo
+	if (_frames[_currentFrame]->_tempo || tempo != _lastTempo) {
 		_puppetTempo = 0;
 	} else if (_puppetTempo) {
 		tempo = _puppetTempo;
 	}
 
 	if (tempo) {
-		if (tempo > 161) {
+		const bool waitForClickOnly = _vm->getVersion() < 300;
+		const int maxDelay = _vm->getVersion() < 400 ? 120 : 60;
+		if (tempo >= 256 - maxDelay) {
```

Comment:
Thank you for your investigation. However, this became complex and fragile. What do you think about adding a small table here enumerating the tempo ranges for every version?